src/visitors/dependencies.js

```diff
@@ -51,6 +54,25 @@ module.exports = {
       node.callee = requireTemplate().expression;
       node.arguments[0] = argTemplate({MODULE: args[0]}).expression;
       asset.isAstDirty = true;
+      return;
+    }
+
+    const isRegisterServiceWorker =
+      callee.property !== undefined &&
+      callee.property.name === 'register' &&
+      callee.object !== undefined &&
+      callee.object.property !== undefined &&
+      callee.object.property.name === 'serviceWorker' &&
+      types.isStringLiteral(args[0]);
+
+    if (isRegisterServiceWorker) {
+      let assetPath = asset.addURLDependency(args[0].value);
+      if (!isURL(assetPath)) {
+        assetPath = urlJoin(asset.options.publicURL, assetPath);
```

Comment:
maybe we should just make `addURLDependency` include the publicURL?

Reply:
not sure- CSSAsset, LESSAsset, SASSAsset and StylusAsset use `addURLDependency`'s return value without prefixing with the public url. Will prefixing with the public url by default cause problems in these asset types?